doc/hardware/peripherals/mspi.rst

```diff
@@ -0,0 +1,197 @@
+   as a parameter to the macro.
+
+#. Initialize the hardware, including but not limited to:
+
+   * check :c:struct:`mspi_cfg` against hardware's own capabilities to prevent
+     incorrect usages.
+
+   * Setup pin mux.
+
+   * Setup the clock for the controller.
+
+   * Power on the hardware.
+
+   * Configure the hardware using :c:struct:`mspi_cfg` and possibly more
+     platform specific settings.
+
+   * Usually, the :c:struct:`mspi_cfg` is filled from device tree and contains
+     no run-time configurable parameters. However, if needed, one can use
+     :c:func:`mspi_config` to re-initialize the hardware with new parameters
+     during run-time.
+
+   * Release any lock if applicable.
+
+#. Perform device driver initialization. As usually, :c:macro:`DEVICE_DT_INST_DEFINE` can be used.
+   Inside device driver initialization function, perform the following required steps.
+
+   #. Call :c:func:`mspi_dev_config` with device specific hardware settings obtained
+      from device datasheets.
+
+      * The :c:struct:`mspi_dev_cfg` should be filled by device tree and helper macro
+        :c:macro:`MSPI_DEVICE_CONFIG_DT` can be used.
+
+      * The controller driver should then validate the members of :c:struct:`mspi_dev_cfg`
+        to prevent incorrect usage.
+
+      * The controller driver should implement a mutex to protect from accidental access.
+
+      * The controller driver may also switch between different devices based on
+        :c:struct:`mspi_dev_id`.
+
+   #. Call API for additional setups if supported by hardware
+
+      * :c:func:`mspi_xip_config` for XIP feature
+
+      * :c:func:`mspi_scramble_config` for scrambling feature
+
+      * :c:func:`mspi_timing_config` for platform specific timing setup.
+
+   #. Reigster any callback with :c:func:`mspi_register_callback` if needed.
+
+   #. Release the controller mutex lock.
+
+Transceive
+===========
+The transceive request is of type :c:struct:`mspi_xfer` which allows for dynamically
+changing the transfer related settings once the mode of operation is determined and
+configured by :c:func:`mspi_dev_config`.
+The API supports bulk transfers with different starting addresses and sizes with
+:c:struct:`mspi_xfer_packet`. However, it is up to the controller implementation
+whether to support scatter IO and callback management. The controller can determine
+which user callback to trigger based on :c:enum:`mspi_bus_event_cb_mask` upon completion
+of each async/sync transfer if the callback had been registered using
+:c:func:`mspi_register_callback`. Or not to trigger any callback at all with
+:c:enum:`MSPI_BUS_NO_CB` even if the callbacks are already registered.
+In which case that a controller supports hardware command queue, user could take the full
+advantage of it in terms of performance if scatter IO and callback management are supported.
+
+Device Tree
+===========
+
+Here is an example for defining a MSPI controller in device tree:
```

Comment:
```suggestion
Here is an example for defining an MSPI controller in device tree:
```

Reply:
I thought we are just supposed to use "an" before vowels. Are you sure?

Reply:
"an" is used before a word that starts with a vowel *sound*, not necessarily just a vowel letter, in this case: "an emm-SPI", just like you would write "an SPI bus", "an FBI agent", etc.

Reply:
Yes, thank you.